core/src/main/java/hudson/FilePath.java

```diff
@@ -1027,8 +1031,15 @@ private boolean installIfNecessaryFrom(@NonNull URL archive, @NonNull TaskListen
             }
 
             long sourceTimestamp = con.getLastModified();
+            String resultEtag = con.getHeaderField("ETag");
```

Comment:
same here

Reply:
Included in df05da5bb5bfde52278bb25f27f77f894a1fd2eb .  Thanks for review.  Could you review again?